wasm/jsapi/limits.any.js

```diff
@@ -0,0 +1,204 @@
+            builder.addGlobal(kWasmI32, true);
+        }
+    });
+
+testLimit("data segments", 1, kJSEmbeddingMaxDataSegments, (builder, count) => {
+        const data = [];
+        builder.addMemory(1, 1, false, false);
+        for (let i = 0; i < count; i++) {
+            builder.addDataSegment(0, data);
+        }
+    });
+
+testLimit("initial declared memory pages", 1, kJSEmbeddingMaxMemoryPages,
+          (builder, count) => {
+            builder.addMemory(count, undefined, false, false);
+          });
+
+testLimit("maximum declared memory pages", 1, kJSEmbeddingMaxMemoryPages,
+          (builder, count) => {
+            builder.addMemory(1, count, false, false);
+          });
+
+testLimit("initial imported memory pages", 1, kJSEmbeddingMaxMemoryPages,
+          (builder, count) => {
+            builder.addImportedMemory("mod", "mem", count, undefined);
+          });
+
+testLimit("maximum imported memory pages", 1, kJSEmbeddingMaxMemoryPages,
+          (builder, count) => {
+            builder.addImportedMemory("mod", "mem", 1, count);
+          });
+
+testLimit("function size", 2, kJSEmbeddingMaxFunctionSize, (builder, count) => {
+        const type = builder.addType(kSig_v_v);
+        const nops = count - 2;
+        const array = new Array(nops);
+        for (let i = 0; i < nops; i++) array[i] = kExprNop;
+        builder.addFunction(undefined, type).addBody(array);
+    });
+
+testLimit("function locals", 1, kJSEmbeddingMaxFunctionLocals, (builder, count) => {
+        const type = builder.addType(kSig_v_v);
+        builder.addFunction(undefined, type)
+          .addLocals({i32_count: count})
+          .addBody([]);
+    });
+
+testLimit("function params", 1, kJSEmbeddingMaxFunctionParams, (builder, count) => {
+        const array = new Array(count);
+        for (let i = 0; i < count; i++) {
+            array[i] = kWasmI32;
+        }
+        const type = builder.addType({params: array, results: []});
+    });
+
+testLimit("function params+locals", 1, kJSEmbeddingMaxFunctionLocals - 2, (builder, count) => {
+        const type = builder.addType(kSig_i_ii);
+        builder.addFunction(undefined, type)
+          .addLocals({i32_count: count})
+          .addBody([kExprUnreachable]);
+    });
+
+testLimit("function returns", 0, kJSEmbeddingMaxFunctionReturns, (builder, count) => {
+        const array = new Array(count);
+        for (let i = 0; i < count; i++) {
+            array[i] = kWasmI32;
+        }
+        const type = builder.addType({params: [], results: array});
+    });
+
+testLimit("initial table size", 1, kJSEmbeddingMaxTableSize, (builder, count) => {
+        builder.setTableBounds(count, undefined);
+    });
+
+testLimit("maximum table size", 1, kJSEmbeddingMaxTableSize, (builder, count) => {
+        builder.setTableBounds(1, count);
+    });
+
+// FIXME(ssauleau): re-enable this test
+testLimit.skip("element segments", 1, kJSEmbeddingMaxElementSegments, (builder, count) => {
```

Comment:
@Ms2ger just for you information, the tests passed here but I had to disable the only very slow test.